src/cc/CMakeLists.txt

```diff
@@ -15,6 +15,12 @@ endif (LIBDEBUGINFOD_FOUND)
 # todo: if check for kernel version
 if (CMAKE_USE_LIBBPF_PACKAGE AND LIBBPF_FOUND)
   include_directories(${LIBBPF_INCLUDE_DIRS})
+  # create up-to-date linux/bpf.h from virtual_bpf.h (remove string wrapper);
+  # when libbpf is built as a submodule we use its version of linux/bpf.h
+  # so this does similar for the libbpf package, removing reliance on the
+  # system uapi header which can be out of date.
+  execute_process(COMMAND sh -c "cd ${CMAKE_CURRENT_SOURCE_DIR}/compat/linux && grep -ve '\\*\\*\\*\\*' virtual_bpf.h > bpf.h")
```

Comment:
This generated `bpf.h` should be added to `.gitignore` ?

Another thought: Maybe we can allow users the specify the path to libbpf UAPI headers, something like:
```shell
$ cmake -DLIBBPF_UAPI_HEADERS_PATH=/path/to/headers ..
```

Reply:
good catch with the .gitignore, thanks! it would be nice to have a configurable path to uapi, but given that we have to keep compat/linux/virtual_bpf.h in sync already, generating a uapi header from it is a useful fallback mechanism for providing a uapi header internally for the bcc build I think. as mentioned below, it will often be the case that there is no up-to-date uapi header available, so I'm hoping we can cater to that case too.